File: app/java/src/main/java/com/example/production/domain/model/quality/LotComposition.java
package com.example.production.domain.model.quality;

import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * ロット構成（トレーサビリティ用）
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class LotComposition {
    private Integer id;
    private String parentLotNumber;
    private String childLotNumber;
    private BigDecimal usedQuantity;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Quality/LotComposition.cs
namespace ProductionManagement.Domain.Models.Quality;

/// <summary>
/// ロット構成
/// </summary>
public class LotComposition
{
    public long? Id { get; init; }
    public required string ParentLotNumber { get; init; }
    public required string ChildLotNumber { get; init; }
    public required decimal UsedQuantity { get; init; }
}
